### PR TITLE
Fix predeploy hooks not running for framework deploy with predeploy hook

### PR DESCRIPTION
**Pull request by freakness**
### Description

Fixes: https://github.com/firebase/firebase-tools/issues/5159

### Scenarios Tested

This fixes deploying a next.js (or other framework) using predeploy hooks in firebase.json.

### CODE REVIEW

File: src/deploy/lifecycleHooks.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ function getChildEnvironment(target: string, overallOptions: any, config: any) {
   let resourceDir;
   switch (target) {
     case "hosting":
-      resourceDir = overallOptions.config.path(config.public);
+      resourceDir = overallOptions.config.path(config.public ?? config.source);
```
Comment (jamesdaniels) [inline src/deploy/lifecycleHooks.ts:51]: Can you add an entry to the changelog?

**Review by jamesdaniels — CHANGES_REQUESTED** [review]

File: src/deploy/lifecycleHooks.ts (hunk 1 above)
Comment (freakness) [reply src/deploy/lifecycleHooks.ts:51]: Done!

**Comment by TheIronDev** [comment]
LGTM.

Unrelated to this PR, lifecycleHooks.ts needs tests.

**Review by TheIronDev — APPROVED** [review]

**Comment by jamesdaniels** [comment]
Merged in #5288, thanks for the fix @freakness

**Pull request closed without merging**